integrations/integrations.go

```diff
@@ -0,0 +1,201 @@
+) (app servertypes.Application, cleanupFn func(), err error) {
+	traceWriter, traceCleanupFn, err := SetupTraceWriter(svrCtx.Logger, svrCtx.Viper.GetString(FlagTraceStore))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	home := svrCtx.Config.RootDir
+	db, err := opts.DBOpener(home, server.GetAppDBBackend(svrCtx.Viper))
+	if err != nil {
+		return app, cleanupFn, err
+	}
+
+	// TODO: Check if is testnet and implement `testnetify` function
+
+	app = appCreator(svrCtx.Logger, db, traceWriter, svrCtx.Viper)
+	cleanupFn = func() {
+		traceCleanupFn()
+		if localErr := app.Close(); localErr != nil {
+			svrCtx.Logger.Error(localErr.Error())
+		}
+	}
+
+	return app, cleanupFn, nil
+}
+
+// See https://github.com/cosmos/cosmos-sdk/blob/7fb26685cd68a6c1d199dc270c80f49f2bfe7ace/server/start.go#L307
+// We modify this function to start a Monomer node in-process instead of a Comet node.
+func startInProcess(
+	svrCtx *server.Context,
+	svrCfg serverconfig.Config,
+	clientCtx client.Context,
+	app servertypes.Application,
+	opts server.StartCmdOptions,
+) error {
+	cmtCfg := svrCtx.Config
+	gRPCOnly := svrCtx.Viper.GetBool(FlagGRPCOnly)
+
+	g, ctx := GetCtx(svrCtx, true)
+
+	if gRPCOnly {
+		svrCtx.Logger.Info("starting node in gRPC only mode; Monomer is disabled")
+		svrCfg.GRPC.Enable = true
+	} else {
+		svrCtx.Logger.Info("starting Monomer node in-process")
+		// Start the Monomer node
+		_, monomerCleanupFn, err := startMonomerNode()
+		if err != nil {
+			return err
+		}
+		defer monomerCleanupFn()
+
+		// Add the tx service to the gRPC router. We only need to register this
+		// service if API or gRPC is enabled, and avoid doing so in the general
+		// case, because it spawns a new local CometBFT RPC client.
+		/* if svrCfg.API.Enable || svrCfg.GRPC.Enable {
+		    clientCtx = clientCtx.WithClient(local.New(tmNode))
+		    app.RegisterTxService(clientCtx)
+		    app.RegisterTendermintService(clientCtx)
+		    app.RegisterNodeService(clientCtx, svrCfg)
+		} */
+	}
+
+	grpcSrv, clientCtx, err := StartGrpcServer(ctx, g, svrCfg.GRPC, clientCtx, svrCtx, app)
+	if err != nil {
+		return err
+	}
+
+	err = StartAPIServer(ctx, g, svrCfg, clientCtx, svrCtx, app, cmtCfg.RootDir, grpcSrv)
+	if err != nil {
+		return err
+	}
+
+	if opts.PostSetup != nil {
+		if err := opts.PostSetup(svrCtx, clientCtx, ctx, g); err != nil {
+			return err
+		}
+	}
+
+	return g.Wait()
+}
+
+// Starts the Monomer node in-process in place of the Comet node.
+func startMonomerNode() (*node.Node, func(), error) {
+	chainID := monomer.ChainID(1)
+	engineWS, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+	cometListener, err := net.Listen("tcp", "127.0.0.1:0")
+	if err != nil {
+		return nil, nil, err
+	}
+	appdb := dbm.NewMemDB()
+
+	// We obviously need to replace this with an actual Monomer Application, not
+	// a testapp.App. How can I do this?
+	app, err := testapp.New(appdb, chainID.HexBig().String())
+	if err != nil {
```

Comment:
We should be able to use the `servertypes.Application` that's in `startInProcess`. We'll probably need to wrap it to implement the `RollbackToHeight` function.

Reply:
I'm a little confused, could you clarify?

I see the `RollbackToHeight` implementation we have in `testapp.go` [here](https://github.com/polymerdao/monomer/blob/81c3603e87322133c9cc73cc5368c816060136bb/testapp/testapp.go#L83). But how do I _wrap_ the `servertypes.Application` to implement this?

Reply:
We need access to the `CommitMultiStore` to rollback (the link above demonstrates that). We may need an ugly type assertion to get access to it... but it might be available some other way.

Reply:
Addressed by creating a wrapper around `servertypes.Application` that reconciles the discrepancies between it and the `monomer.Application` interface.